src/boot/bootctl-status.c

```diff
@@ -130,6 +174,35 @@ static int print_efi_option(uint16_t id, int *n_printed, bool in_order) {
 
         efi_tilt_backslashes(path);
 
+        if (arg_json_format_flags != JSON_FORMAT_OFF) {
+                _cleanup_(json_variant_unrefp) JsonVariant *efi_obj = NULL;
+                char id_buffer[10];
+                snprintf(id_buffer, sizeof(id_buffer), "0x%04X", id);
+                char status_buffer[100];
+                snprintf(status_buffer, sizeof(status_buffer), "%sactive%s", active ? "" : "in", in_order ? ", boot-order" : "");
+                char partition_buffer[100];
```

Comment:
please do not use buffers with made-up sizes. either use asprintf() (for astuff that is dynamic in size). Or use DECIMAL_STR_MAX() and similar macros to size the buffer the exact appropriate size. In the latter case use xsnprintf() to ensure we abort  if the size calculation doesn't match the limits we assuped.